open_src/influx/coordinator/statement_executor.go

```diff
@@ -489,30 +491,107 @@ func (e *StatementExecutor) executeCreateMeasurementStatement(stmt *influxql.Cre
 	if err := meta2.ValidShardKey(stmt.ShardKey); err != nil {
 		return err
 	}
+
 	e.StmtExecLogger.Info("create measurement ", zap.String("name", stmt.Name))
+	if _, err := e.MetaClient.Measurement(stmt.Database, stmt.RetentionPolicy, stmt.Name); err != meta2.ErrMeasurementNotFound {
+		return meta2.ErrMeasurementExists
+	}
+
+	schema := make(map[string]int32)
+	for _, tag := range stmt.Tags {
+		if _, ok := schema[tag]; !ok {
+			schema[tag] = 0
+		} else {
+			return fmt.Errorf("field or tag conflict %s", tag)
+		}
+	}
+
+	for field, _ := range stmt.Fields {
+		if _, ok := schema[field]; !ok {
+			schema[field] = 0
+		} else {
+			return fmt.Errorf("field or tag conflict %s", field)
+		}
+	}
+
 	ski := &meta2.ShardKeyInfo{ShardKey: stmt.ShardKey, Type: stmt.Type}
 	indexR := &meta2.IndexRelation{}
-	if len(stmt.IndexList) > 0 {
-		for i, indexType := range stmt.IndexType {
-			oid, err := tsi.GetIndexIdByName(indexType)
+
+	if len(stmt.IndexInfo) > 0 {
+		var textIndexs []*meta2.IndexInfor
+		var fieldIndexs []*meta2.IndexInfor
+		for _, IndexInfo := range stmt.IndexInfo {
+			if IndexInfo.IndexType == "keyword" {
+				IndexInfo.IndexType = "field"
+			}
+
+			oid, err := tsi.GetIndexIdByName(IndexInfo.IndexType)
 			if err != nil {
 				return err
 			}
-			if oid == uint32(tsi.Field) && len(stmt.IndexList[i]) > 1 {
-				return fmt.Errorf("cannot create field index for multiple columns: %v", stmt.IndexList[i])
+			if _, ok := stmt.Fields[IndexInfo.FieldName]; !ok {
+				return fmt.Errorf("%s index only create on field, but not field: %s", IndexInfo.IndexType, IndexInfo.FieldName)
+			}
+
+			index := &meta2.IndexInfor{
+				FieldName:  IndexInfo.FieldName,
+				Tokens:     IndexInfo.Tokens,
+				Tokenizers: IndexInfo.Tokenizers,
+				IndexName:  IndexInfo.IndexName,
+			}
+
+			if oid == uint32(tsi.Field) {
+				fieldIndexs = append(fieldIndexs, index)
+			} else if oid == uint32(tsi.Text) {
+				if val, _ := stmt.Fields[IndexInfo.FieldName]; val == influx.Field_Type_String {
+					textIndexs = append(textIndexs, index)
+					continue
```

Comment:
This "continue" is no effect.  In addtion，the type of fields that need to crate field index and text index is influx.Field_Type_String. Not only Text index.